pkgs/development/python-modules/pex/default.nix

```diff
@@ -2,23 +2,26 @@
   lib,
   buildPythonPackage,
   fetchPypi,
-  hatchling,
+  setuptools,
```

Comment:
So, you're already super-fragile and have to re-do dep grabbing by hand in your build process (by design I know). As such, it would seem uv could be fetched in this manual list of deps and then maybe some sort of Pex build plumbing could be made aware of this fact... just thinking out loud.

Reply:
Ah - right, no good. The uv distribution is fake just a binary blob. So this won't work.

Reply:
Sort of; dep grabbing is cached after the package has been successfully built by the build infra behind nixpkgs.
So an approach [like the one suggested here](https://github.com/NixOS/nixpkgs/pull/382636#discussion_r1957416102) would ultimately yield a built wheel containing the exes.
Users of this package would not have to go through the build process again afaik.

Reply:
A related topic is that the pex_build build system is unsuited to nix, due to its fetching of dependencies during build time.
A nix-compatible solution would be to create a package for the pex_build package so we can plug it into the `build-system` field. This package would then be responsible for fetching the EXEs, instead of the packages it is meant to be building

Reply:
Great - sounds like you have this under control. Thanks for filling me in a bit on the details. From a tool maintainer perspective re-packagers are a strange case (few of them) vs "normal" users. Making it clear the gymnastics were coming from an effort to re-package is super-good up-front info for things like https://github.com/pex-tool/pex/issues/2687.

Reply:
But I fear that would be too much for this PR; @phaer is mostly interested in removing themselves as maintainer, I imagine simply fetching the EXEs and revisiting this topic further down the line would be more appropriate :)

Reply:
I won't linger to long on the ~obvious contradiction though of a-ok to fetch 4 blobs to build Pex from sources but not ok to just fetch the Pex sdist or whl blobs from PyPI. I can't see how one is much better than another from a security exposure perspective.

Reply:
> A nix-compatible solution would be to create a package for the pex_build package.

So publish my private one off build system to PyPI?

Reply:
So-called nix fetchers require passing a hash (usually sha256) to verify that the file has not changed between downloads.
This does, of course, assume that the initial hash represents a safe binary, but otherwise prevents attacks that involve substituting the binaries

Reply:
> This does, of course, assume that the initial hash represents a safe binary, but otherwise prevents attacks that involve substituting the binaries

But you could simply do that for the pex whl itself directly, right? Or the sdist, from which you could build the whl. The sdist both has the Pex sources and the binary blobs, all in one shot.

Reply:
> So publish my private one off build system to PyPI?
Sounds a bit like overkill; if it's a one off then there's no reason to do so, as packages with few usages aren't desired. If there were further packages that used this, then it could be considered.

Reply:
Ok. So you're aware, these are a general and official thing in the Python world: https://peps.python.org/pep-0517/#in-tree-build-backends